modules/aws/files/bootstrap_role_iam_policy.json.tpl

```diff
@@ -89,7 +89,8 @@
       "Sid": "ResR53Z",
       "Effect": "Allow",
       "Action": [
-        "route53:ChangeResourceRecordSets"
+        "route53:ChangeResourceRecordSets",
+        "route53:DeleteHostedZone"
```

Comment:
Add this because when provisioning environment with SN-managed DNS but then change to Customer-managed DNS, at that scenario, our pipeline will fail since unable to clean the already provisioned Zone.